datajunction-clients/python/datajunction/client.py

```diff
@@ -50,58 +51,59 @@ def list_metrics(self, namespace: Optional[str] = None) -> List[str]:
             )
         return self._session.get("/metrics/").json()
 
+    def list_cubes(self, namespace: Optional[str] = None) -> List[str]:
+        """
+        List metric nodes for given namespace or all.
+        TODO
+        """
+
     #
     # Get common metrics and dimensions
     #
-    def common_dimensions_with_details(
+    def common_dimensions(
         self,
         metrics: List[str],
-    ) -> List[dict]:  # pragma: no cover # Tested in integration tests
+        name_only: bool = False,
+    ) -> List[Union[str, dict]]:  # pragma: no cover # Tested in integration tests
         """
         Return common dimensions for a set of metrics.
         """
         query_params = []
         for metric in metrics:
             query_params.append((models.NodeType.METRIC.value, metric))
-        response = self._session.get(
+        json_response = self._session.get(
             f"/metrics/common/dimensions/?{urlencode(query_params)}",
-        )
-        return response.json()
-
-    def common_dimensions(
-        self,
-        metrics: List[str],
-    ) -> List[str]:  # pragma: no cover # Tested in integration tests
-        """
-        Return common dimensions (names only) for a set of metrics.
-        """
-        dimensions = self.common_dimensions_with_details(metrics=metrics)
-        return [dimension["name"] for dimension in dimensions]
+        ).json()
+        if name_only:
+            return [dimension["name"] for dimension in json_response]
+        return json_response
 
-    def common_metrics_with_details(
+    def common_metrics(
         self,
         dimensions: List[str],
-    ) -> List[dict]:  # pragma: no cover # Tested in integration tests
+        name_only: bool = False,
+    ) -> List[Union[str, dict]]:  # pragma: no cover # Tested in integration tests
         """
         Return common metrics for a set of dimensions.
         """
         query_params = [("node_type", models.NodeType.METRIC.value)]
         for dim in dimensions:
             query_params.append((models.NodeType.DIMENSION.value, dim))
-        response = self._session.get(
+        json_response = self._session.get(
             f"/dimensions/common/?{urlencode(query_params)}",
-        )
-        return response.json()
-
-    def common_metrics(
-        self,
-        dimensions: List[str],
-    ) -> List[str]:  # pragma: no cover # Tested in integration tests
-        """
-        Return common metrics (names only) for a set of dimensions.
-        """
-        metrics = self.common_metrics_with_details(dimensions=dimensions)
-        return [metric["name"] for metric in metrics]
+        ).json()
+        if name_only:
+            return [metric["name"] for metric in json_response]
+        return [
+            {
```

Comment:
Would it be better here to return a list of `Metric` objects rather than a dict? Or is the idea that this would eventually provide `paths` or other DAG information, thus making it incompatible with a `Metric` object?

Reply:
Yes, exactly, I think this return type should be minimal, similarly to the dimension side. I think adding paths to the dimensions would also be interesting here.